qiskit/primitives/base/base_primitive.py

```diff
@@ -72,3 +74,38 @@ def _cross_validate_circuits_parameter_values(
         return validation._cross_validate_circuits_parameter_values(
             circuits, parameter_values=parameter_values
         )
+
+
+BasePrimitive = BasePrimitiveV1
+
+
+class BasePrimitiveV2(ABC):
+    """Primitive abstract base class version 2."""
+
+    version = 2
+    _options_class: type[BasePrimitiveOptions] = BasePrimitiveOptions
+
+    def __init__(self, options: Optional[BasePrimitiveOptionsLike] = None):
+        self._options: type(self._options_class)
+        self._set_options(options)
+
+    @property
+    def options(self) -> BasePrimitiveOptions:
+        """Options for BaseEstimator"""
+        return self._options
+
+    @options.setter
+    def options(self, options: BasePrimitiveOptionsLike):
+        self._set_options(options)
+
+    def _set_options(self, options):
```

Comment:
I dont think this method is necessary. You could just have something like this in the init:
```python
self._options = self._options_class()
if options:
    self._options.update(options)
```
and then any coercian has to be handled by the update method. eg it could handle a mapping , **mapping` or another options class (which isn't a mapping)